source/computer/macos-setup.rst

```diff
@@ -1,9 +1,213 @@
+可以用于安装各种常见的软件包、库文件以及字体。日常及科研工作中所需的大多数软件、
+库文件以及字体都可以通过 Homebrew 安装。
+
+要安装 Homebrew，需要将如下命令复制到终端中并执行::
+
+    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+
+安装好以后，打开新的 Terminal 便可以使用 ``brew`` 命令。
+Homebrew 的详细用法见官方文档。此处仅列出一些常用的用法::
+
+    # 模糊搜索与 wget 相关的软件
+    $ brew search wget
+
+    # 安装 wget 软件包
+    $ brew install wget
+
+    # 安装 Visual Studio Code
+    # 由于 Visual Studio Code 是带图形界面的软件，因而这里需要使用 --cask 选项
+    $ brew install --cask visual-studio-code
+
+    # 升级某个软件
+    $ brew upgrade xxx
+
+    # 卸载某个软件
+    $ brew uninstall xxx
+
+.. note::
+
+   Homebrew 可能在国内下载速度较慢，可以考虑使用中科大 LUG 提供的 Homebrew 镜像。
+   详细使用方法见：
+
+   - Homebrew 源镜像：http://mirrors.ustc.edu.cn/help/brew.git.html
+   - Homebrew Bottles 源镜像：http://mirrors.ustc.edu.cn/help/homebrew-bottles.html
+   - Homebrew Core 源镜像：http://mirrors.ustc.edu.cn/help/homebrew-core.git.html
+   - Homebrew Cask 源镜像：http://mirrors.ustc.edu.cn/help/homebrew-cask.git.html
+
+编程开发环境
+------------
+
+C
+^
+
+Command Line Tools for Xcode 已经提供了 C 编译器和相关工具，因而无需单独安装
+C编译器。
+
+Fortran
+^^^^^^^
+
+若需要编译 Fortran 程序，则可以安装 `GNU Fortran <https://gcc.gnu.org/fortran/>`__
+编译器（即 ``gfortran``\ ）::
+
+    $ brew install gfortran
+
+Java
+^^^^
+
+若要运行 Java 程序，则需要安装 Java 开发环境::
+
+    $ brew install openjdk
+    $ sudo ln -sfn /usr/local/opt/openjdk/libexec/openjdk.jdk /Library/Java/JavaVirtualMachines/openjdk.jdk
+
+Python
+^^^^^^
+
+macOS 默认的 Python 版本为 Python 2.7，而目前主流的 Python 版本为
+Python 3.x。建议通过 `Anaconda <https://seismo-learn.org/software/anaconda/>`__
+来安装最新版 Python。
+
+git
+^^^
+
+`git <https://git-scm.com/>`__ 是最流行的版本控制工具，也是科研过程中编写代码
+以及项目管理推荐使用的软件。使用如下命令安装 git::
+
+    $ brew install git
+
+.. note::
+
+   使用 ``brew`` 命令安装的 git 位于 ``/usr/local/bin/`` 目录下，而
+   Command Line Tools for Xcode 携带的 git 位于 ``/usr/bin/``
+   目录下。我们使用 ``echo $PATH`` 命令查看环境变量 ``$PATH`` 会发现，
+   一般情况下，\ ``/usr/local/bin/`` 位于 ``/usr/bin/`` 前。因此，
+   此时我们使用的 git 其实是通过 ``brew`` 安装的版本。
+
+其他命令行工具
+--------------
+
+这里推荐一些常用的命令行工具。
+
+Windows 和 Linux/macOS 系统下，文本文件的换行符时不同的。
+dos2unix 可以将 Windows 系统下的换行符转换为 UNIX 系统下的换行符。
+unix2dos 则反之::
+
+    $ brew install dos2unix unix2dos
+
+`gawk <https://www.gnu.org/software/gawk/>`__ 是用于文本处理的命令行工具。
+macOS 自带的 awk 语法上与 gawk 有所不同。推荐安装并使用 gawk::
+
+    $ brew install gawk
+
+`wget <https://www.gnu.org/software/wget/>`__ 是用于下载文件的命令行工具::
+
+    $ brew install wget
+
+其他效率软件
+------------
+
+如上所说，Homebrew 可以用于安装日常及科研的大部分软件。这里推荐一些可以提升
+效率的 macOS 软件。
+
+
+自 macOS Catalina 开始，系统默认的 Shell 由 bash 变为 zsh。
+`Oh My Zsh <https://github.com/ohmyzsh/ohmyzsh>`__ 一套 zsh 配置文件，
+非常方便使用。使用如下命令安装::
+
+    $ sh -c "$(curl -fsSL https://raw.github.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"
+
+该命令下载 Oh My Zsh 到 ``~/.oh-my-zsh`` 目录下，生成默认的 zsh 配置文件 ``.zshrc``\ ，
+并备份配置文件老的配置文件到 ``.zshrc.pre-oh-my-zsh``\ 。
+
+根据自身习惯，修改 ``.zshrc``\ 。比如设置喜欢的主题，如 ``bira``，安装喜欢的插件，
+如 ``sudo``\ 、``extract``\ 、``autojump``\ 、``zsh-autosuggestions``\ 、
+``zsh-syntax-highlighting`` 等。
```

Comment:
Perhaps we should move oh-my-zsh to "best practices". I think it's definitely better than bash, and should be used by all Linux and macOS users.

Reply:
Yes, I agree.